src/main/kotlin/no/nav/familie/ks/sak/kjerne/behandling/SnikeIKøenService.kt

```diff
@@ -0,0 +1,175 @@
+import no.nav.familie.ks.sak.kjerne.logg.LoggService
+import org.slf4j.LoggerFactory
+import org.springframework.stereotype.Service
+import org.springframework.transaction.annotation.Transactional
+import java.time.Clock
+import java.time.LocalDateTime
+
+@Service
+class SnikeIKøenService(
+    private val behandlingService: BehandlingService,
+    private val loggService: LoggService,
+    private val tilbakestillBehandlingService: TilbakestillBehandlingService,
+    private val clock: Clock
+) {
+    private val logger = LoggerFactory.getLogger(javaClass)
+
+    @Transactional
+    fun settAktivBehandlingPåMaskinellVent(
+        behandlingId: Long,
+        årsak: SettPåMaskinellVentÅrsak,
+    ) {
+        val behandling = behandlingService.hentBehandling(behandlingId)
+
+        if (!behandling.aktiv) {
+            throw IllegalStateException("Behandling=$behandlingId er ikke aktiv")
+        }
+
+        val behandlingStatus = behandling.status
+        val behandlingStegTilstand = finnBehandlingStegTilstand(behandling)
+
+        val erBehandlingIkkePåVent = behandlingStegTilstand.behandlingStegStatus !== BehandlingStegStatus.VENTER
+
+        if (erBehandlingIkkePåVent && behandlingStatus !== BehandlingStatus.UTREDES) {
+            throw IllegalStateException("Behandling=$behandlingId kan ikke settes på maskinell vent då status=$behandlingStatus")
+        }
+
+        behandling.status = BehandlingStatus.SATT_PÅ_MASKINELL_VENT
+        behandling.aktiv = false
+        behandlingService.oppdaterBehandling(behandling)
+        loggService.opprettSettPåMaskinellVent(behandling, årsak.beskrivelse)
+    }
+
+    private fun finnBehandlingStegTilstand(behandling: Behandling) =
+        behandling.behandlingStegTilstand.single { it.behandlingSteg == behandling.steg }
+
+    /**
+     * @param behandlingSomFerdigstilles er behandlingen som ferdigstilles i [no.nav.familie.ba.sak.kjerne.steg.FerdigstillBehandling]
+     *  Den er mest brukt for å logge hvilken behandling det er som ferdigstilles og hvilken som blir deaktivert
+     *
+     * @return boolean som tilsier om en behandling er reaktivert eller ikke
+     */
+    @Transactional
+    fun reaktiverBehandlingPåMaskinellVent(behandlingSomFerdigstilles: Behandling): Reaktivert {
+
+        val behandlingerPåFagsak = behandlingService.hentBehandlingerPåFagsak(behandlingSomFerdigstilles.fagsak.id)
+
+        val behandlingPåMaskinellVent = finnBehandlingPåMaskinellVent(behandlingerPåFagsak) ?: return Reaktivert.NEI
+
+        val behandlingStegTilstand = finnBehandlingStegTilstand(behandlingPåMaskinellVent)
+
+        val erBehandlingStegStatusVenter = behandlingStegTilstand.behandlingStegStatus == BehandlingStegStatus.VENTER
+
+        val aktivBehandling = behandlingerPåFagsak.singleOrNull { it.aktiv }
+
+        validerBehandlinger(aktivBehandling, behandlingPåMaskinellVent)
+
+        aktiverBehandlingPåVent(aktivBehandling, behandlingPåMaskinellVent, behandlingSomFerdigstilles)
```

Comment:
Kan vurdere å trekke det ut i to metoder ala `deaktiverAktivBehandling(aktivBehandling)` og `aktiverBehandlingPåMaskinellvent(behandlingPåMaskinellVent)` (litt usikker på om man egentlig trenger å sende inn `behandlingSomFerdigstilles` her)